airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -0,0 +1,196 @@
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class VaultSecrets(BaseSecretsBackend, LoggingMixin):
+    """
+    Retrieves Connection object from Hashicorp Vault Secrets
+
+    Configurable via ``airflow.cfg`` as follows:
+
+    .. code-block:: ini
+
+        [secrets]
+        backend = airflow.providers.hashicorp.secrets.vault.VaultSecrets
+        backend_kwargs = {"path": "airflow", "url": "http://127.0.0.1:8200"}
+
+    For example, if your keys are under ``airflow/AIRFLOW_CONN_SMTP_DEFAULT``, this would be accessible if you
+    provide ``{"path": "airflow"}`` and request conn_id ``smtp_default``.
+
+    :param path: Specifies the path of the secret to read.
+    :type path: str
+    :param url: Base URL for the Vault instance being addressed.
+    :type url: str
+    :param auth_type: Authentication Type for Vault (one of 'token', 'ldap', 'userpass', 'approle',
+        'github', 'gcp). Default is ``token``.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. (Default: ``secret``)
+    :type mount_point: str
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param username: Username for Authentication (for ``ldap`` and ``userpass`` auth_type)
+    :type username: str
+    :param password: Password for Authentication (for ``ldap`` and ``userpass`` auth_type)
+    :type password: str
+    :param role_id: Role ID for Authentication (for ``approle`` auth_type)
+    :type role_id: str
+    :param secret_id: Secret ID for Authentication (for ``approle`` auth_type)
+    :type secret_id: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+    :type gcp_key_path: str
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        path: str,
+        url: Optional[str] = None,
+        auth_type: str = 'token',
+        mount_point: str = 'secret',
+        token: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        role_id: Optional[str] = None,
+        secret_id: Optional[str] = None,
+        gcp_key_path: Optional[str] = None,
+        gcp_scopes: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__()
+        self.path = path.rstrip('/')
+        self.url = url
+        self.auth_type = auth_type
+        self.kwargs = kwargs
+        self.token = token
+        self.username = username
+        self.password = password
+        self.role_id = role_id
+        self.secret_id = secret_id
+        self.mount_point = mount_point
+        self.gcp_key_path = gcp_key_path
+        self.gcp_scopes = gcp_scopes
+        self._client: Optional[hvac.Client] = None
+
+    def get_client(self) -> hvac.Client:
```

Comment:
```suggestion
    @cached_property
    def client(self) -> hvac.Client:
```

And then we don't need `self._client` or the `if` in this fn.

Reply:
Updated